src/PowerShellEditorServices.Engine/Services/Analysis/AnalysisService.cs

```diff
@@ -673,6 +691,232 @@ private class PowerShellResult
 
             public bool HasErrors { get; }
         }
+
+        internal async Task RunScriptDiagnosticsAsync(
+            ScriptFile[] filesToAnalyze)
+        {
+            // If there's an existing task, attempt to cancel it
+            try
+            {
+                if (s_existingRequestCancellation != null)
```

Comment:
We may need a semaphore around this logic, but I think leave til later. Worth putting a comment in